jaeger-jaxrs2/src/test/java/com/uber/jaeger/propagation/ServerFilterIntegrationTest.java

```diff
@@ -0,0 +1,197 @@
+package com.uber.jaeger.propagation;
+
+import static com.uber.jaeger.Constants.DEBUG_ID_HEADER_KEY;
+import static com.uber.jaeger.Constants.DEBUG_JAEGER_SPAN_BAGGAGE_HEADER;
+import static org.junit.Assert.assertEquals;
+
+import com.uber.jaeger.context.ActiveSpanSourceTraceContext;
+import com.uber.jaeger.context.TraceContext;
+import com.uber.jaeger.filters.jaxrs2.ClientFilter;
+import com.uber.jaeger.filters.jaxrs2.ServerFilter;
+import com.uber.jaeger.reporters.NoopReporter;
+import com.uber.jaeger.samplers.ConstSampler;
+import io.opentracing.Span;
+import io.opentracing.Tracer;
+import java.io.IOException;
+import java.net.URI;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import org.glassfish.grizzly.http.server.HttpServer;
+import org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpServerFactory;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.test.JerseyTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ServerFilterIntegrationTest extends JerseyTest {
+
+  // Base URI the Grizzly HTTP server will listen on
+  private static final String BASE_URI = "http://localhost:8080/";
```

Comment:
just curious - is it possible to specify port 0 and let the OS pick an ephemeral port? This is how we write tests in Go, because hardcoding a specific port, especially a commonly used one like 8080, could make the tests unstable/sensitive to environment.